File: 268.missing-number.py
#
# @lc app=leetcode id=268 lang=python3
#
# [268] Missing Number
#

# @lc code=start
from typing import *
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        bucket = list(range(len(nums)+1))
        for num in nums:
            bucket[num] = -1
        for num in bucket:
            if not num == -1:
                return num
        
        
# @lc code=end


if __name__ == "__main__":
    print(Solution().missingNumber([1,4,5,6,3]))

File: 268.missing-number.go
/*
 * @lc app=leetcode id=268 lang=golang
 *
 * [268] Missing Number
 */
package main

import "fmt"

//  bit-manipulation
// @lc code=start
func missingNumber(nums []int) int {
	n := len(nums)
	ret := 0
	for i := 1; i <= n; i++ {
		ret ^= i
	}
	for _, num := range nums {
		ret ^= num
	}
	return ret

}

// @lc code=end
func main() {
	fmt.Println(missingNumber([]int{3, 0, 1}))
}
